Assets/Scripts/InteractionDetection.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+public class InteractionDetection : MonoBehaviour
+{
+    private Queue<GameObject> _intObjPull;
+    private GameObject intObj;
+    private bool _interInput = false;
+    private InteractionScript _is;
+    [SerializeField] public GameObject player;
```

Comment:
нет смысла делать public filed с [SerializeField] attribute
```suggestion
    [SerializeField] private GameObject player;
```